docs/src/pages/components/checkboxes/Checkboxes.js

```diff
@@ -33,13 +33,6 @@ export default function Checkboxes() {
           'aria-label': 'disabled checked checkbox',
         }}
       />
-      <Checkbox
-        defaultChecked
-        indeterminate
```

Comment:
I'd still prefer this here. I understand the standalone demo but we usually have all possible states in the initial demo. I think that's important when evaluating a design system.

Reply:
I disagree, we need to break this demo down into smaller chunks, at least: size. Moving out color and disabled into standalone demos could also be interesting. cc @mbrookes for his perspective.

Reply:
If you look how design systems present themselves then you always see a collection of base use cases. I would want that as well. I don't want to have to read through every demo to find the state I was looking for. Maybe we don't need them in the first example but some example should have all states for evaluation.

Reply:
> I don't want to have to **read through every demo to find the state I was looking for**.

- If you already know what you are looking for, I think that you would be able to run a textual search to find the content. Breaking the demo help on this matter. We should require to read through every demo.
- If you don't already know what you are looking for, I think that breaking the demos open the opportunity to give more context around use state, which should help too.

**Now, on the opposite side**, if we have a big demo with all the base use cases, we make it so developers have to open the source and read through each component until they find the one that they have seen on the screen, with no to little way to speed this process.

> see a collection of base use cases.

In the case of material.io, they [start with](https://material.io/components/selection-controls):

<img width="564" alt="Capture d’écran 2020-09-19 à 18 21 59" src="https://user-images.githubusercontent.com/3165635/93671650-08a80b80-faa5-11ea-94f2-a4290ca58c9d.png">

I think that it would make a great replacement for this demo: https://material-ui.com/components/checkboxes/#basic-checkboxes. I don't think that disabled, size, color, indeterminate, etc would quality as a base use case. I would be leaning toward saying that the base use case is only: **unchecked** and **checked**.

> we usually have all possible states in the initial demo

I think we should move away from the current demos that show it all. #22484: the inline preview isn't the end goal, but a metric that we are close to simpler demos.

> I think that's important when evaluating a design system.

On this one, I have no idea how team evaluates a design system. There is no way a team can look at everything in depth, they would never have the time. If I had to guess, I imagine that on average, they look at 3 aspects with equal weight:

1. the design, does it look & feel great, can I override it, can I use Material Design?
2. the features, does it has all the components I will need?
3. everything else.

Reply:
I have tried to summarize the tradeoff in https://twitter.com/olivtassinari/status/1307361081950699523 without creating biases in any direction.

Reply:
> If you already know what you are looking for,

The point is that I don't know what I'm looking for or rather "where are all the possible states?". Especially since the state in question as an ambiguous name: indeterminate and mixed.

Reply:
You didn't ask the question I was talking about. You mixed in "disabled" like it is a standalone feature of a component. It is not. It is a possible state the component is in. So I don't think you want to discuss the same thing. And since other's are now twisting it more into "all features" I don't see this as a constructive discussion anymore.

Reply:
I have undone the change.